<!-- Launching all required nodes for playing Tic Tac Toe with Baxter -->
<launch>

    <include file="$(find baxter_tictactoe)/launch/board_sensor.launch" />

    <!-- Range thresholds for the red tokens -->
    <rosparam param = "ttt_controller/hsv_red">
        H: [160,  20]
        S: [ 40, 196]
        V: [ 50, 196]
    </rosparam>

    <!-- Range thresholds for the blue tokens -->
    <rosparam param = "ttt_controller/hsv_blue">
        H: [ 60, 130]
        S: [ 90, 256]
        V: [ 10, 256]
    </rosparam>

    <rosparam param = "ttt_controller/tile_pile_position">[0.52, 0.83, -0.09]</rosparam>

    <!-- 3D positions of the corners of the board -->
    <rosparam param = "ttt_controller/board_corner_poss">
        TL: [0.74, 0.63, -0.18]
        TR: [0.77, 0.30, -0.18]
        BR: [0.45, 0.29, -0.18]
        BL: [0.45, 0.63, -0.18]
    </rosparam>

    <!-- Objects database for the left arm -->
    <rosparam param = "ttt_controller/objects_left">
        "no_obj" : -1
        "tile_1" :  1
        "tile_2" :  2
        "tile_3" :  3
        "tile_4" :  4
        "tile_5" :  5
        "tile_6" :  6
        "tile_7" :  7
        "tile_8" :  8
        "tile_9" :  9
    </rosparam>

    <rosparam param="/print_level">3</rosparam>
    <rosparam param="ttt_controller/num_games">3</rosparam>
    <rosparam param="ttt_controller/cheating_games">[2, 3]</rosparam>
    <!-- <param name="ttt_controller/voice"       type="str" value="voice_cmu_us_jmk_arctic_clunits" /> -->
    <param name="ttt_controller/voice"       type="str" value="voice_kal_diphone" />
    <param name="ttt_controller/robot_color" type="str" value=             "red" />

    <node name="tictactoe_brain" pkg="baxter_tictactoe" type="tictactoe_brain" respawn="false" output="screen" required="true"/>

    <node name="soundplay_node" pkg="sound_play" type="soundplay_node.py" required="true"/>
</launch>
